backend/core/src/email/email.service.ts

```diff
@@ -207,7 +207,15 @@ export class EmailService {
     const jurisdiction = await this.getUserJurisdiction(user)
     void (await this.loadTranslations(jurisdiction, user.language))
     const compiledTemplate = this.template("forgot-password")
-    const resetUrl = `${appUrl}/reset-password?token=${user.resetToken}`
+    const urlObj = new URL(appUrl)
+
+    const redirectUrl = urlObj.searchParams.get("redirectUrl")
+    const listingId = urlObj.searchParams.get("listingId")
+
+    const resetUrl =
+      redirectUrl && listingId && process.env.SHOW_MANDATED_ACCOUNTS
```

Comment:
Nit: You have this pattern in a few places, and it's a very nit comment on performance, but best practice is you generally want to have toggles checked first in a list like this - we don't need to evaluate any logic following a toggle if the toggle is false.